app/views/devise/shared/_auth_nav.html.slim

```diff
@@ -0,0 +1,10 @@
+.clearfix.mxn1.mb1
+  - login_page = controller_name == 'sessions'
+  .sm-col.sm-col-6.px1.mb1
+    = link_to t('upaya.links.sign_up'), new_user_registration_path, \
+      class: 'btn btn-primary border-box col-12 center ' \
+             "#{login_page ? 'bg-gray' : 'bg-navy'}"
+  .sm-col.sm-col-6.px1.mb1
+    = link_to t('upaya.links.sign_in'), new_user_session_path, \
+      class: 'btn btn-primary border-box col-12 center ' \
+             "#{login_page ? 'bg-navy' : 'bg-gray'}"
```

Comment:
For now, I think you're right. The logic is not too complex. I thought that it might also be easier to test, but it looks like the way you have it is easily testable. Can you please add tests for both of these links in the sessions#new and registrations#new view specs?

You can test that each link has the appropriate class by specifying the controller like this in the `before` block:
```ruby
allow(view).to receive(:controller_name).and_return('sessions')
```
and then test for the link href and class like this:
```ruby
sign_up_class = 'btn btn-primary border-box col-12 center bg-gray'
expect(rendered).
  to have_xpath("//a[@class='#{sign_up_class}' and @href='#{new_user_registration_path}']")
```